### DISCUSSION TITLE
How do I set up Google Adsense ads?

### DISCUSSION THREAD

**Discussion by linuxmobile** [uPagge/uBlogger#68] [Q&A] [answered]
### Describe the bug

_I recent found this new theme based on LoveIt, and i like too mucho. This have more features and updates than LoveIt._ I want to **add and ads advertising**, like your latest release. **But i can't find where to add that lines!**

### Build Environment

- OS: [Archlinujx]
- Theme version/commit [2.0.2]
- Hugo version [latest]

**Comment by uPagge** [comment, marked as answer]
To begin with, this is not a template problem. Therefore, such questions need to be raised in discussions)

At the moment, you can only add ads from Google.
This is done through configuration.

```
params:
  advertising:
    enable: true
    google:
      client: ca-pub-xxxxxxxx
      slot: xxxxxxxxxxx
```

These are the default settings.

The advertisement itself is inserted into arbitrary places of the template using the shortcode: {{<ads>}} 
It also has parameters:
slot - changes the default ad unit to another, useful if you have several authors on your blog, you can create an ad unit for each of them
circuit - a boolean field, determines whether the ad should be surrounded by a frame

**Reply by linuxmobile** [reply]
Well, thx u for help with it! sorry if i post on wrong place!

**Reply by uPagge** [reply]
It's okay, please contact if you have any questions))
